xdsl/ir.py

```diff
@@ -15,12 +15,41 @@
 OpT = TypeVar('OpT', bound='Operation')
 
 
+@dataclass
+class Dialect:
+    """Contains the operations and attributes of a specific dialect"""
+    __operations: list[Operation] = field(default_factory=list,
+                                          init=True,
+                                          repr=True)
+    __attributes: list[Attribute] = field(default_factory=list,
+                                          init=True,
+                                          repr=True)
+
+    def get_operations(self) -> Iterator[Operation]:
+        return iter(self.__operations)
+
+    def get_attributes(self) -> Iterator[Attribute]:
+        return iter(self.__attributes)
+
+    def __call__(self, ctx: MLContext) -> None:
```

Comment:
This function allows backward compatibility. Do we want that here?

Reply:
I would say that we shouldn't keep it, especially since we do not have many users yet.
What do you think @georgebisbas ? You know more than us about these things.

Reply:
Keeps compatibility for what exactly?

Reply:
Previously, we registered dialects with `_ = Arith(ctx)`. This still allows to register dialects like this, hence keeping backward compatibility.